src/errors/index.ts

```diff
@@ -91,3 +92,56 @@ export class FileNotFoundError extends Error {
     this.name = FileNotFoundError.name;
   }
 }
+
+/**
+ * Type guard to check if an error is an Octokit RequestError.
+ *
+ * This function checks the structure of the error object to determine if it matches
+ * the shape of a RequestError. It should be favored instead of `instanceof` checks,
+ * especially in scenarios where the prototype chain might not be reliable, such as when
+ * dealing with different versions of a package or when the error object might have been
+ * modified.
+ *
+ * @param error The error object to check.
+ * @returns A boolean indicating whether the error is a RequestError.
+ */
+export function isOctokitRequestError(error: unknown): error is RequestError {
+  if (typeof error === 'object' && error !== null) {
+    const e = error as RequestError;
+    return (
+      e.name === 'HttpError' &&
+      typeof e.status === 'number' &&
+      typeof e.request === 'object'
```

Comment:
question: is there anything that's more octokit-specific that we could check here as well?

Reply:
Not that I can see.

```ts
import { RequestOptions, ResponseHeaders, OctokitResponse } from "@octokit/types";
import { RequestErrorOptions } from "./types";
/**
 * Error with extra properties to help with debugging
 */
export declare class RequestError extends Error {
    name: "HttpError";
    /**
     * http status code
     */
    status: number;
    /**
     * http status code
     *
     * @deprecated `error.code` is deprecated in favor of `error.status`
     */
    code: number;
    /**
     * Request options that lead to the error.
     */
    request: RequestOptions;
    /**
     * error response headers
     *
     * @deprecated `error.headers` is deprecated in favor of `error.response.headers`
     */
    headers: ResponseHeaders;
    /**
     * Response object if a response was received
     */
    response?: OctokitResponse<unknown>;
    constructor(message: string, statusCode: number, options: RequestErrorOptions);
}

```

Reply:
Good point, I don't think there's much chance of this causing issues anyway, seems like a fairly low surface area.